spacy_transformers/layers/transformer_model.py

```diff
@@ -211,12 +211,14 @@ def backprop_transformer(d_output: FullTransformerBatch) -> List[Doc]:
 
 def _convert_transformer_inputs(model, wps: WordpieceBatch, is_train):
     # Adapter for the HFWrapper. See https://thinc.ai/docs/usage-frameworks
+
+    hf_device = model.shims[0]._hfmodel.transformer.device
```

Comment:
I see your point. But could the user have the transformer on a different device from the one that's current? That would produce the wrong result here, right?

Reply:
You are right, it's better to explicitly use the device of the transformer. Then I'll leave it as it is.

It's probably better to have a general solution for wrapping anyway, since `tokenizer` and the other properties that use `model.shims[0]._hfmodel.transformer` also break with wrapping?

Reply:
Hmm. Why does this break though, if I think about it?

The shim should still be attached to this model instance, right? The outer wrapper should end up calling into this. Did we not construct a new `Model` for the wrapping function?

Reply:
After recursively wrapping `Model`s, `self.layers[0]` is a wrapper, so the shim is at `self.layers[0].layers[0].shims[0]` rather than `self.layers[0].shims[0]._hfmodel.tokenizer`.

Reply:
Now that I think of it, it is surprising it doesn't work, since the wrapper's first layer is set to be the wrapped `Model`'s first layer. I'll look what goes wrong exactly when I am at the office/GPU machine.

Reply:
Ah, it currently isn't that explains it :), I'll update https://github.com/explosion/spacy-transformers/pull/289.

Reply:
Done, now it works without issues.